pkg/controller/portworxdiag/pod.go

```diff
@@ -0,0 +1,135 @@
+		{Name: "dockersock", MountPath: "/var/run/docker.sock"},
+		{Name: "containerddir", MountPath: "/run/containerd"},
+		{Name: "criosock", MountPath: "/var/run/crio"},
+		{Name: "crioconf", MountPath: "/etc/crictl.yaml"},
+		{Name: "optpwx", MountPath: "/opt/pwx"},
+		{Name: "procmount", MountPath: "/host_proc"},
+		{Name: "sysdmount", MountPath: "/etc/systemd/system"},
+		{Name: "dbusmount", MountPath: "/var/run/dbus"},
+		{Name: "varlibosd", MountPath: "/var/lib/osd", MountPropagation: &mp},
+		{Name: "diagsdump", MountPath: "/var/cores"},
+		{Name: "etcpwx", MountPath: "/etc/pwx"},
+		{Name: "journalmount1", MountPath: "/var/run/log", ReadOnly: true},
+		{Name: "journalmount2", MountPath: "/var/log", ReadOnly: true},
+	}
+}
+
+func tolerations() []v1.Toleration {
+	return []v1.Toleration{
+		{Key: "node.kubernetes.io/not-ready", Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoExecute},
+		{Key: "node.kubernetes.io/unreachable", Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoExecute},
+		{Key: "node.kubernetes.io/disk-pressure", Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoSchedule},
+		{Key: "node.kubernetes.io/memory-pressure", Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoSchedule},
+		{Key: "node.kubernetes.io/pid-pressure", Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoSchedule},
+		{Key: "node.kubernetes.io/unschedulable", Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoSchedule},
+		{Key: "node.kubernetes.io/network-unavailable", Operator: v1.TolerationOpExists, Effect: v1.TaintEffectNoSchedule},
+	}
+}
+
+func makeDiagPodTemplate(diag *portworxv1.PortworxDiag, ns string, nodeName string, nodeID string) (*v1.PodTemplateSpec, error) {
+	svcLinks := true
+	terminationGP := int64(10)
+	privileged := true
+
+	diagImage := "docker.io/portworx/oci-monitor"
+	diagTag := "akrpantest" // TODO: change once px-installer PR is merged
+
+	if img := os.Getenv("DIAG_IMAGE"); len(img) > 0 {
+		diagImage = img
+	}
+	if tag := os.Getenv("DIAG_IMAGE_TAG"); len(tag) > 0 {
+		diagTag = tag
+	}
+
+	return &v1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "portworx-diag",
+			Namespace:    ns,
+			Labels: map[string]string{
+				"name": PortworxDiagLabel,
+			},
+		},
+		Spec: v1.PodSpec{
+			NodeName:                      nodeName,
+			HostPID:                       true, // We do need this
+			HostNetwork:                   true, // Do we need this?
+			RestartPolicy:                 v1.RestartPolicyOnFailure,
+			DNSPolicy:                     v1.DNSClusterFirst,
+			EnableServiceLinks:            &svcLinks, // Do we need this?
+			ServiceAccountName:            serviceAccountName,
+			TerminationGracePeriodSeconds: &terminationGP,
+			Tolerations:                   tolerations(),
+			Volumes:                       volumes(),
+			Containers: []v1.Container{
+				{
+					Name:            "diag-collector",
+					Image:           fmt.Sprintf("%s:%s", diagImage, diagTag),
```

Comment:
Let's use the custom registry specified in the storagecluster. All the other components use this method to add the image registry to the image - https://github.com/libopenstorage/operator/blob/master/pkg/util/util.go#L169


Similarly we need to add the image pull secret if it is specified in the storagecluster spec.